Catch hover on as const

kunaljaydesai opened the pull request:
Fixes #43515 

First PR here, would appreciate some help!

kunaljaydesai commented on src/compiler/checker.ts:
```diff
@@ -15636,6 +15636,11 @@ namespace ts {
+                    if (node.kind === SyntaxKind.Identifier as TypeNodeSyntaxKind &&
+                        node.parent.kind === SyntaxKind.TypeReference &&
+                        (<Identifier>(node as Node)).originalKeywordKind === SyntaxKind.ConstKeyword) {
+                        return errorType;
```
This isn't the right type to return, what would be?

Kingwl commented:
I think this is not an issue of type checker.

May consider about `src\services\services.ts@getNodeForQuickInfo`. (Add some special case here).

kunaljaydesai commented:
thanks for the pointers @Kingwl! I've updated the PR to reflect that and added a test

another option to consider is maybe `isConstTypeReference(node.parent)` in `shouldGetType`:
<img width="916" alt="Screen Shot 2021-04-07 at 11 38 41 PM" src="https://user-images.githubusercontent.com/7613905/113979668-665e3400-97fa-11eb-9ecc-1d0c1e818a18.png">

let me know which is preferred!

Kingwl commented:
I felt ok with both of them options.

You could modify the `getNodeForQuickInfo` If you wanna provide some quick info (eg: `node as type` shows node if you hover to type).

But If you want to keep it silent, `shouldGetType` might be a better choice.

kunaljaydesai commented:
I think I'll keep it silent for now so opted for the second approach! Let me know if there's anything else needed to get the PR merged

Kingwl commented on tests/cases/fourslash/quickInfoForConstTypeReference.ts:
```diff
@@ -0,0 +1,5 @@
+
+////"" as /**/const;
+
+verify.quickInfoAt("", "");
```
Use `verify.not.quickInfoExists()` instead.

kunaljaydesai replied:
updated!

Kingwl commented:
BTW: It's no need to force push. We usually merge the PR with squash.

Kingwl reviewed (approved):
Looks good! Thanks for your contribution.

orta reviewed (approved):
Agree, thanks!
